File: android/src/main/java/com/loylty/plugins/elitesdk/RedemptionKitIonicHelperPlugin.java
package com.loylty.plugins.elitesdk;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "RedemptionKitIonicHelper")
public class RedemptionKitIonicHelperPlugin extends Plugin {

    private RedemptionKitIonicHelper implementation = new RedemptionKitIonicHelper();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }
}

File: ios/Sources/RedemptionKitIonicHelperPlugin/RedemptionKitIonicHelperPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(RedemptionKitIonicHelperPlugin)
public class RedemptionKitIonicHelperPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "RedemptionKitIonicHelperPlugin"
    public let jsName = "RedemptionKitIonicHelper"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = RedemptionKitIonicHelper()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
